v4: Homepage, navbar, examples updates

mdo opened the pull request:
This PR brings v4's docs homepage closer in line to the upcoming v5 docs changes, featuring a cleaner and brighter homepage and some updated styles elsewhere.

- Updates navbar to use the brighter purple, making things feel livelier.
- Updates purple docs buttons to no longer be outlined.
- Update homepage masthead content, add Icons section, redo Themes section, and more with new layout.
- Update examples index page to streamline buttons in the header and add Themes callout at the bottom.

Woo, hoping to get this into v4.4.2. Meant to open the PR yesterday, but I lost all my changes when I was working too fast. Think I've captured it all here though.

Preview: <https://deploy-preview-30769--twbs-bootstrap.netlify.app/>

TODO:

- [x] Add `loading=lazy`
- [x] Make sure the images have the maximum dimensions they are going to be displayed and that they are not scaled down via the browser
- [x] Compress all the images
- [x] Fix tap targets are not sized properly Lighthouse error (not related to this PR, we fixed in on master, I'll backport it - XhmikosR)
- [x] Resize `bootstrap-themes-collage.png` to 1150px width; now it's 1151px (I will re-compress it after that)
- [x] Verify the other changes

XhmikosR commented:
I'll have a look first things tomorrow :) BTW you can skip any dist files like docs.min.js

XhmikosR commented on site/index.html:
```diff
@@ -33,57 +26,96 @@ <h1 class="mb-3 bd-text-purple-bright">Bootstrap</h1>
-{% highlight sh %}
-gem install bootstrap -v {{ site.current_ruby_version }}
+# Via gem
+gem install bootstrap -v 4.4.1
```
this seems wrong

XhmikosR commented:
@mdo please always compress the images. I will improve compression tomorrow, but currently bootstrap-themes@2x.png is ~2MB :/ Also, perhaps try with jpeg?

mdo commented:
Agreed on image size. I’ll take another pass today or tomorrow on sizing and make sure they’re appropriate on pixel dimensions and then compare jpg/png.

XhmikosR commented:
I updated my comment. The only think we should do first is make sure the 1x images have the maximum dimensions they will be displayed. And adapt the 2x images after that.

mdo commented:
Updated and re-compressed the images across the board—tons of file size savings now. Should be good for you tomorrow to review and merge at your leisure @XhmikosR. Thanks!

mdo committed:
v4: Update homepage, examples, and general docs styles

- Refreshes the look and feel of the homepage with fresh colors, new content, and simplified styles.
- Updates docs button styles
- Replaces bytesize icons with Bootstrap icons
- Adds themes callout to examples page

XhmikosR committed:
Add `loading="lazy"` in images

Backport of #30199 after adapting it for v4-dev.

XhmikosR committed:
Backport #29272 and adapt if for v4-dev

docs: fix tap targets

XhmikosR commented:
@mdo I pushed my patches.

Things left:

- [ ] Resize `bootstrap-themes-collage.png` to 1150px width; now it's 1151px (I will re-compress it after that)
- [ ] There's one small tap target warning in navbar's `<a class="nav-link px-1 mx-1 py-3 my-n2" href="https://github.com/twbs" target="_blank" rel="noopener" aria-label="GitHub">...`
- [ ] I'm not sure I like the copy buttons not showing in all homepage snippets. Especially for the second snippet. For the first, I'm not sure how to proceed since now they are one snippet
- [ ] If we really want to hide the copy buttons, ideally, we shouldn't add the copy button instead of hiding it. But that's mostly nitpicking, although it does cause a slight page jump. Not a blocking issue, just something I might address later, and probably applies to master too.

mdo commented:
> * I'm not sure I like the copy buttons not showing in all homepage snippets. Especially for the second snippet. For the first, I'm not sure how to proceed since now they are one snippet

Code snippets are already like that on the homepage. Agreed on if we could not load them there, but that's also something we can revisit in the future or correct with v5.

MartijnCuppens commented:
Wow, this looks cool, nice work @mdo!
